### PR TITLE
Feat/circular modals

### PR DESCRIPTION
**Pull request by babadrape**
- Add Sell, Buy, Pool Harvest and General Harvest Modals
- Add Modals to swap (when selling and buying banana)
- Add General Harvest Modal to farms when unstaking, harvesting and harvesting all
- Add General Harvest  Modal to maximizers when unstaking, harvesting and harvesting all
- Add General Harvest Modal to bills pages when claiming Banana bills
- Add Pool Harvest to pools page when harvesting and unstaking from banana pools

### CODE REVIEW

File: src/state/user/reducer.ts [hunk 1]
```diff
@@ -90,6 +91,13 @@ export interface UserState {
   watchlistPools: string[]
   showPhishingWarningBanner: boolean
   unlimitedGnana: boolean
+
+  showModal: {
+    selling: boolean
+    buying: boolean
+    poolHarvest: boolean
+    generalHarvest: boolean
```
Comment (Harambe-Nakamoto) [inline src/state/user/reducer.ts:99]: Lets use more descriptive vars because this will grow

File: src/utils/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { BigNumber } from '@ethersproject/bignumber'
 import { getAddress } from '@ethersproject/address'
-import { BLOCK_EXPLORER } from 'config/constants/chains'
+import { BLOCK_EXPLORER, CHAIN_ID } from 'config/constants/chains'
```
Comment (Harambe-Nakamoto) [inline src/utils/index.ts:3]: Use chainId from SDK

File: src/state/user/actions.ts [hunk 3]
```diff
@@ -66,4 +66,9 @@ export const setIsExchangeChartDisplayed = createAction<boolean>('user/toggleIsE
 
 export const setUnlimitedGnana = createAction<boolean>('user/setUnlimitedGnanaMinting')
 
+export const toggleModal = createAction<{
+  actionType: string
+  value: boolean
```
Comment (Harambe-Nakamoto) [inline src/state/user/actions.ts:71]: can you change this variable name to flag

File: src/state/user/reducer.ts (hunk 1 above)
Comment (babadrape) [reply src/state/user/reducer.ts:99]: Does something like "sellModal" work?

File: src/state/user/actions.ts [hunk 4]
```diff
@@ -66,4 +66,9 @@ export const setIsExchangeChartDisplayed = createAction<boolean>('user/toggleIsE
 
 export const setUnlimitedGnana = createAction<boolean>('user/setUnlimitedGnanaMinting')
 
+export const flagModal = createAction<{
+  actionType: string
+  value: boolean
```
Comment (Harambe-Nakamoto) [inline src/state/user/actions.ts:71]: I meant this param: value to flag. You can keep the previous name for the action

**Pull request merged**